components/action-mgt/org.wso2.carbon.identity.action.execution/src/main/java/org/wso2/carbon/identity/action/execution/ActionExecutorServiceImpl.java

```diff
@@ -396,18 +408,20 @@ private void logErrorResponse(Action action, ActionInvocationErrorResponse error
 
     private void logFailureResponse(Action action, ActionInvocationFailureResponse failureResponse) {
 
-        if (LOG.isDebugEnabled()) {
-            // todo: add to diagnostic logs
+        if (LOG.isDebugEnabled() || LoggerUtils.isDiagnosticLogsEnabled()) {
```

Comment:
Let's remove this if condition as it checks the same conditions again inside the if

Reply:
Should take the diaganotic log outside the if and if should be applicable only for the system debug log

Reply:
Updated in